backend/src/main/controller/PromotionController.ts

```diff
@@ -164,4 +164,50 @@ export class PromotionController {
       return next(e);
     }
   };
+
+  /**
+   * Upvotes a promotion
+   */
+  upVotePromotion = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    try {
+      await getManager().transaction(async (transactionalEntityManager) => {
+        const id = await IdValidation.schema.validateAsync(request.params.id, {
+          abortEarly: false,
+        });
+        await transactionalEntityManager
+          .getCustomRepository(PromotionRepository)
+          .increment({ id: id }, 'votes', 1);
```

Comment:
```suggestion
          .increment({ id }, 'votes', 1);
```

Super minor improvement here but also not a big deal!